packages/crayons-core/src/components/icon/icon.tsx

```diff
@@ -1,51 +1,154 @@
-import { Component, Prop, State, Watch, h } from '@stencil/core';
-
-// Icons Object
-import icons from './icon-assets/icons.json';
+import {
+  Build,
+  Component,
+  Element,
+  Prop,
+  State,
+  Watch,
+  h,
+} from '@stencil/core';
+import { fetchIcon } from '@freshworks/crayons-icon/icon-utils/icon-fetch-memoize';
 
 @Component({
   tag: 'fw-icon',
   styleUrl: 'icon.scss',
   assetsDirs: ['icon-assets'],
   shadow: true,
 })
+/**
+ * Class of the icon. Suppoprts Intersection Observer, Icon Cache, JS Imports for Tree shaking inline SVGs and SVGs from external CDNs or custom path within project.
+ */
 export class Icon {
   /**
-   * Identifier of the icon. The attribute’s value must be a valid svg file in the repo of icons (assets/icons).
+   * Identifier of the icon. The attribute’s value must be a valid svg Name in the Crayons-Icon set.
    */
   @Prop() name: string;
-
   /**
-   * Size of the icon, specified in number of  pixels.
-   * Default value is 12px defined using the --icon-size css variable.
+   * Identifier of the icon. The attribute’s value must be a valid JS Import Name of the svg in the named export from @freshworks/crayons-icon.
    */
-  @Prop() size: number;
-
+  @Prop() crayonIcon = '';
```

Comment:
can we rename it to `crayonsIcon` instead of `crayonIcon`

Reply:
fixed. closed.

Reply:
In the latest pull, I still the prop name as `crayonIcon` instead of `crayonsIcon`